### PR TITLE
DDF-2745: Fix resource-uri being overriden by null update

### PR DESCRIPTION
**Pull request by peterhuffer**
#### What does this PR do?
- Fix resource-uri being overriden by null update

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@emmberk @troymohl @brendan-hofmann 

#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Security](https://github.com/orgs/codice/teams/security)

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@pklinef
@stustison

#### How should this be tested? (List steps with links to updated documentation)
Full Build.

#### Any background context you want to provide?
Existing `Metacard`'s URIs should not be able to be overridden in order to prevent resources with no references to it.

#### What are the relevant tickets?
[DDF-2745](https://codice.atlassian.net/browse/DDF-2745)
#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/plugin/catalog-plugin-content-uri/src/main/java/org/codice/ddf/catalog/content/plugin/uri/ContentUriAccessPlugin.java [hunk 1]
```diff
@@ -41,19 +45,31 @@ public UpdateRequest processPreUpdate(UpdateRequest input,
...
                     && ContentItem.CONTENT_SCHEME.equals(existingMetacard.getResourceURI()
                     .getScheme())) {
-                throw new StopProcessingException("Cannot overwrite resource URI in content store.");
+                throw new StopProcessingException(CANNOT_OVERWRITE_MESSAGE);
             }
         }
         return input;
     }
 
+    private boolean oneOrOtherUriNull(Metacard existingMetacard, Metacard updatedMetacard) {
```
Comment (brendan-hofmann) [inline catalog/plugin/catalog-plugin-content-uri/src/main/java/org/codice/ddf/catalog/content/plugin/uri/ContentUriAccessPlugin.java]: I think this check should also include the URI scheme check, as this plugin is only meant for content URIs.

Comment (pklinef) [inline catalog/plugin/catalog-plugin-content-uri/src/main/java/org/codice/ddf/catalog/content/plugin/uri/ContentUriAccessPlugin.java]: I would suggest doing the content scheme check for everything in the `for` loop which would include this method call.

Comment (peterhuffer) [inline catalog/plugin/catalog-plugin-content-uri/src/main/java/org/codice/ddf/catalog/content/plugin/uri/ContentUriAccessPlugin.java]: Went ahead and added this check at the beginning of the `for` loop before checking anything else. Also added an additional test where both existing and updated URIs are content schema and equal.

**Comment by pklinef** [comment]
Make sure to either squash your second commit or add the JIRA ticket to the commit message.

**Comment by peterhuffer** [comment]
Squashed commits.

**Comment by peterhuffer** [comment]
I think merging this as it is now will break `UpdateStorageRequest`s for `Metacard`s that already have a URI because we remove the URI from the incoming `ContentItem`'s `Metacard` for security reasons, so it will always be null. Need to figure out what to do about this.

**Comment by peterhuffer** [comment]
I have removed where we filter out the `RESOURCE_URI` inside the `OverrideAttributeSupport` and also added two integration tests, both for `CreateStorageRequests` and `UpdateStorageRequests` to ensure the `RESOURCE_URI` cannot be overridden.

File: catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java [hunk 2]
```diff
@@ -715,7 +715,7 @@ public Response updateDocument(@PathParam("id") String id, @Context HttpHeaders
                 if (multipartBody != null) {
                     List<Attachment> contentParts = multipartBody.getAllAttachments();
                     if (contentParts != null && contentParts.size() > 0) {
-                        createInfo = parseAttachment(contentParts.get(0));
+                        createInfo = parseAttachments(contentParts, transformerParam);
```
Comment (peterhuffer) [inline catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java]: The `updateDocument` RESTEndpoint was not correctly parsing all attachments of the incoming request.

File: catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/OverrideAttributesSupportTest.java [hunk 3]
```diff
@@ -63,7 +63,7 @@ public void testOverrideAttributesBasic() throws URISyntaxException {
                 .getTitle(), is("updated"));
         assertThat(metacardMap.get("original")
                 .getResourceURI()
-                .toString(), is("content:stuff"));
```
Comment (brendan-hofmann) [inline catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/OverrideAttributesSupportTest.java]: These test lines are a little bit weird now. They would make more sense if they were something other than a content uri, like an ftp uri or something.

Comment (peterhuffer) [inline catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/OverrideAttributesSupportTest.java]: It shouldn't matter what the URI is. It is asserting that the URI is unchanged.

Comment (brendan-hofmann) [inline catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/OverrideAttributesSupportTest.java]: I understand it has no impact on the test results, but it is confusing/misleading to anyone who tries to look at our unit tests for examples of how the software should be used or should behave. 😺

**Pull request merged**